util/dvsim/Testplan.py

```diff
@@ -119,6 +120,15 @@ def _validate(self):
                              f"invalid:\n{self}\nLegal values: "
                              f"Testpoint.milestones")
 
+        # "tests" key must be list.
+        if not isinstance(self.tests, list):
+            raise ValueError(f"'tests' key in {self} is not a list.")
+
+        # "tags" key if present, must be a list.
+        if hasattr(self, "tags"):
+            if not isinstance(self.tags, list):
+                raise ValueError(f"'tags' key in {self} is not a list.")
```

Comment:
Can I suggest assigning `self.tags` to [] or None if it doesn't exist? That way, we won't need to use `hasattr` after running the `_validate()` method.